### PR TITLE
[All hosts] (code samples) Add code samples

### PR DESCRIPTION
**Pull request by samantharamon**
Adds code samples to Office.UI.addHandlerAsync and Office.AsyncContextOptions.

### CODE REVIEW

File: generate-docs/script-inputs/local-repo-snippets.yaml [hunk 1]
```diff
@@ -1798,6 +1798,39 @@ Office.AddinCommands.Source#id:member:
 
         event.completed();
     }
+Office.AsyncContextOptions:interface:
+  - |-
+    // The following example gets the attachment contents of the
+    // current mail item being composed in Outlook.
+    function getAttachmentContentCompose() {
+        const item = Office.context.mailbox.item;
+        const options = { asyncContext: { currentItem: item } };
```
Comment (AlexJerabek) [inline generate-docs/script-inputs/local-repo-snippets.yaml:1807]: For these class-level samples, we should make it very clear what the object is.
```suggestion
        const options: Office.AsyncContextOptions = { asyncContext: { currentItem: item } };
```

File: generate-docs/script-inputs/local-repo-snippets.yaml [hunk 2]
```diff
@@ -1798,6 +1798,39 @@ Office.AddinCommands.Source#id:member:
...
+        item.getAttachmentsAsync(options, callback);
+
+        function callback(result) {
```
Comment (AlexJerabek) [inline generate-docs/script-inputs/local-repo-snippets.yaml:1810]: How much of this function is required to know how the callback options work? I don't mind the full sample, but I wonder if the function contents could be `//...` and get the same point across. (Though maybe I'm oversimplifying this and missing something.)

**Review by AlexJerabek — APPROVED** [review]
Approving with a couple suggestions.

File: generate-docs/script-inputs/local-repo-snippets.yaml (hunk 2 above)
Comment (samantharamon) [reply generate-docs/script-inputs/local-repo-snippets.yaml:1810]: I initially went with that option, but determined that the comments didn't help simplify the sample. Having the full callback function still helped to show how to store callback options and use them.

**Pull request merged**